tests/FileDetector.php

```diff
@@ -216,69 +216,53 @@ public function TryDeduceEngine( array $Files, array $Matches ) : ?string
 			//This is a really long and annoying check. Basically we have two things to look for:
 			//1. A single .pck file named exactly "data.pck", and NO other pck files
 			//2. For every executable, a correspondingly named pck file, and no other pck files
-			
-			$HasDataPck = false;
-			
+
 			$Pcks = [];
 			$Exes = [];
-			
-			$FoundExe = false;
-			$FoundApp = false;
-			$FoundX86 = false;
-			$FoundX64 = false;
-			
-			//There's also the case where a linux executable has no extension but I am not bothering to mess with that
-			
-			$ExePckPairs = 0;
-			
+
 			foreach( $Files as $File )
 			{
 				$Extension = pathinfo( $File, PATHINFO_EXTENSION );
 				$BaseFile = basename($File);
-				
+
 				if( $Extension === 'exe' )
 				{
-					$FoundExe = true;
 					$Exes[ $BaseFile ] = $swapExtension($BaseFile, ".exe", ".pck");
 				}
 				else if( $Extension === 'app' )
 				{
-					$FoundApp = true;
 					$Exes[ $BaseFile ] = $swapExtension($BaseFile, ".app", ".pck");
 				}
 				else if( $Extension === 'x86' )
 				{
-					$FoundX86 = true;
 					$Exes[ $BaseFile ] = $swapExtension($BaseFile, ".x86", ".pck");
 				}
 				else if( $Extension === 'x86_64' )
 				{
-					$FoundX64 = true;
 					$Exes[ $BaseFile ] = $swapExtension($BaseFile, ".x86_64",".pck");
 				}
 				else if( $Extension === 'pck' )
 				{
 					$Pcks[ $BaseFile ] = $BaseFile;
-					if($BaseFile === "data.pck") $HasDataPck = true;
 				}
+
+				//There's also the case where a linux executable has no extension but I am not bothering to mess with that
 			}
-			
+
 			//If we have exactly 1 PCK file and it is data.pck, we can skip all the fancy checks
-			if(count($Pcks) === 1 && $HasDataPck)
+			if( count( $Pcks ) === 1 && array_key_first($Pcks) === 'data.pck' )
 			{
 				return 'Engine.Godot';
 			}
-			
+
 			//Otherwise we have to match up exe & pck pairs
-			$Pairs = [];
-			
 			foreach($Exes as $exe) {
-				
+
 				//If we have found a particular exe format, ensure there is a correspondingly named PCK file.
 				unset( $Pcks[ $exe ] );
-				
+
 			}
-			
+
 			//Make sure we do not have any "orphan" pck files that aren't paired with an executable
 			//There are some Godot games like that, but it's not worth the false positives
```

Comment:
I'm not sure I agree :) It would be useful to see a list of potential matches which are excluded by this, to see if we have more potential false positives (if any) than potential false negatives (now or in the future).

So far I only saw `data.pck` + other random PCKs in false positives, not `<exe_name>.pck` + other PCKs.

Reply:
If you want to test for yourself, you can modify the script and add some tests , just add like so:

```
types/twopass/Engine.Godot.<APPID>.txt
```

Then just run `php test tests/Test.php`

Otherwise I'll get around to fiddling with this, but it might not be today